src/sumo_gym/envs/fmp.py

```diff
@@ -79,58 +79,115 @@ def __init__(
 
         else:
             (
-                raw_vertices,  # id, x, y
-                # raw_charging_stations,
-                # raw_electric_vehicles,
-                raw_edges,  # edge_id, id_start, id_dest
-                raw_departures,  # vehicle_id, edge_id_from
-                raw_demand,  # start_id, dest_id
+                raw_vertices,  # [id (str), x_coord (float), y_coord (float)]
+                raw_charging_stations, # [id, (x_coord, y_coord), edge_id, charging speed]
+                raw_electric_vehicles, # [id (str), maximum speed (float), maximumBatteryCapacity (float)]
+                raw_edges,  # [id (str), from_vertex_id (str), to_vertex_id (str)]
+                raw_departures,  # [vehicle_id, starting_edge_id]
+                raw_demand,  # [junction_id, dest_vertex_id]
             ) = sumo_gym.utils.xml_utils.decode_xml_fmp(
-                net_xml_file_path, demand_xml_file_path
+                net_xml_file_path, demand_xml_file_path,
+                additional_xml_file_path
             )
 
-            # todo: better implement vertex_dict using np.idx
-            vertices = []
-            self.vertex_dict = {}
+            self.vertices = []
```

Comment:
Here I used vertices rather than self.vertices, because in mypy python type checking, a variable has a static type. Means you cannot change self.vertices from list to np.ndarray. And to match the data struct in data, we use ndarray.

Reply:
so Shuo means you should use "vertices = []" when appending and extracting, and finally do 

> self.vertices = np.asarray(vertices)

same to all other self. variables, like edges